src/v/raft/state_machine.cc

```diff
@@ -198,8 +198,8 @@ state_machine::insert_linearizable_barrier(
      * Inject leader barrier and wait until returned offset is applied
      */
 
-    return _raft->linearizable_barrier(timeout).then(
-      [this, timeout](result<model::offset> r) {
+    return _raft->linearizable_barrier(timeout)
```

Comment:
unrelated to this PR, wondering if this method should hold the stm gate.

Reply:
We lack this in pretty much all methods of all STMs. I have it on my list to fix.